plugin.yaml

```diff
@@ -10,62 +10,318 @@ plugins:
     # the plugin if it is included in the the blueprint directory under the
     # "plugins" directory.
-    source: https://github.com/cloudify-cosmo/cloudify-gcp-plugin/archive/master.zip
+    source: https://github.com/szpotona/cloudify-gcp-plugin/archive/add_disk.zip
 
 node_types:
   cloudify.gcp.nodes.Instance:
     derived_from: cloudify.nodes.Compute
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists and use existing (true) 
+          or if Cloudify should create new resource (false).
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the GCP API.
+        default: {}
+      image_id:
+        description: >
+          The ID of the image in your GCP account.
+        type: string
+        default: {}
+      instance_type:
+        description: >
+          The instance's type. All available instance types can be found here: 
+          https://cloud.google.com/compute/docs/machine-types
+        type: string
+        default: n1-standard-1
+      instance_additional_settings:
+        description: >
+          Additional instance settings.
+        default: {}
     interfaces:
       cloudify.interfaces.lifecycle:
         start:
           implementation: gcp_plugin.plugin.tasks.create_instance
           inputs:
-            config:
-              default: {}
-        stop:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config]}
+            properties:
+              default: { get_property: [SELF, instance_additional_settings]}
+            instance_type:
+              default: { get_property: [SELF, instance_type]}
+            image_id:
+              default: { get_property: [SELF, image_id]}
+        delete:
           implementation: gcp_plugin.plugin.tasks.delete_instance
           inputs:
-            config:
-              default: {}
+            gcp_config:
+              default: { get_property: [SELF, gcp_config] }
+
+  cloudify.gcp.nodes.Volume:
+    derived_from: cloudify.nodes.Volume
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      image:
+        description: >
+          The image of the Volume.
+        default: ''
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: gcp_plugin.plugin.tasks.create_disk
+          inputs:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config] }
+            image:
+              default: { get_property: [SELF, image] }
```

Comment:
naming is a bit confusing here, because the instance also has an image, which is named 'image_id'. I think this property should be renamed to 'disk_image' (or 'disk_image_id' of the format is ID)